### DISCUSSION TITLE
Compiling Topola Viewer with Visual Studio Code

### DISCUSSION THREAD

**Discussion by vejrous** [PeWu/topola-viewer#50] [Q&A]
Hi,
so I tried to compile project in Visual Studio Code (Typescript is completly new to me) and got theese step together:

**1. Download Node.js**
from _https://nodejs.org/en/_ and install it.

**2. Run Visual studio code**

**2.1 Install Typescript** compiler in VS Code by running following command in Terminal:
_npm install -g typescript_

You can test your install by checking the version or help:
_tsc --version
tsc --help_

**2.2 Open project** directory from VS Code menu:
_File_ -> _Open Folder_ -> select folder with Topola project

**2.3 Installing all dependencies**
Project has a package.json file. Dependencies will be install by running in Terminal:
_npm install_

It will install everything the project needs, in the node_modules folder, creating it if it's not existing already.

**2.4 Run from Terminal** (have no idea what does it do):
_npm i --save-dev @types/jest_

**2.5 Compiling** from menu VS Code menu:
_Terminal_ -> _Run Build Task_ (Ctrl+Shift+B)
_tsc: build - tsconfig.json_

**But I end with 2 errors:**
> Executing task: c:\Dokumenty\Projects\Topola\node_modules\.bin\tsc.cmd -p c:\Dokumenty\Projects\Topola\tsconfig.json <
>
> src/intro.tsx:3:18 - error TS2307: Cannot find module './topola.jpg' or its corresponding type declarations.
> import logo from './topola.jpg';
>
> src/menu/wikitree_menu.tsx:3:26 - error TS2307: Cannot find module './wikitree.png' or its corresponding type declarations.
> import wikitreeLogo from './wikitree.png';

**Comment by PeWu** [comment]
This should be fixed with the latest commit (see #51).
I have only used `npm run buld` to build the project. Now it should work with running `tsc` directly.
